### PR TITLE
Test the CLI

### PR DESCRIPTION
**Pull request by ajthinking**
### Background
* Sometimes the GUI interacts with a JS server directly in browser via a LocalClient
* Sometimes the GUI makes HTTP requests to a remote node app via RemoteClient. (This way the core can be extended to  provide private/custom nodes for a specific business/domain problem.
* :point_right: **Sometimes the GUI makes HTTP requests to remote _server in language X_..**

For this case core is bundled to a "CLI" for "portable execution". For instance a PHP/Laravel app might use this CLI like this
```php
DataStory::run($diagram)
```
Under the hood PHP calls `node lib/scr/cli/cli.js run <SERIALIZED_DIAGRAM>`

Similarly it may do `DataStory::boot()` to provide available nodes to PHP which then forwards it to the gui who asked...

### Why not use the JS version in frontend in these cases?
* We may want to schedule diagram executions in PHP app.
* A future feature we want to implement is to provide a _context_ to execution. Then Language X can pass data to the CLI - something like `DataStory::context(['prices' => $prices)->run($priceListUpdaterDiagram)`.

### This PR
* Fixes the build. (`tsc` builds for web then `npx webpack` uses those files to bundle the cli to a single file)
* Adds a test. Unfortunately the cli relies on `stdoutput` causing `yarn test` to output lots of things to console. This can be avoided with `yarn test --silent`
* builds lib in github-actions

### Links
[Demo PHP app using CLI](https://github.com/ajthinking/price-app/tree/master/app/DataStory) (slightly outdated)

### CODE REVIEW

**Comment by ajthinking** [comment]
@Lenivaya,
* Can you take a look at this PR and provide any feedback or questions
* We recently ignored the `lib` dir. How does gui consume those files then?
* Also we need a place to put the cli. What do you think? lib or a dist folder or attach to release like this maybe
image

**Comment by Lenivaya** [comment]
> * We recently ignored the `lib` dir. How does gui consume those files then?

Keeping build artifacts in source control is really not a best practice but I think due to the specifics of the project we can tell webpack to compile and leave cli in /cli/cli.js. We also need to include cli folder in eslint and prettier ignore-files so as not to do extra work by checking a lot of code.
> * Also we need a place to put the cli. What do you think? lib or a dist folder or attach to release like this maybe

Great idea would be to create a workflow with automatic release creation for every push on master, or maybe pushed tags. Will take care of this after finishing rewriting gui.

File: webpack.config.js [hunk 1]
```diff
@@ -2,10 +2,10 @@ const path = require('path');
 
 module.exports = {
   target: 'node',
-  entry: './lib/cli/data-story.js',
+  entry: './lib/src/cli/data-story.js',
   mode: 'production',
   output: {
     filename: 'cli.js',
-    path: path.resolve(__dirname, 'lib/cli'),
+    path: path.resolve(__dirname, 'lib/src/cli'),
```
Comment (Lenivaya) [inline webpack.config.js:9]: Chaning path to 
```js
path: path.resolve(__dirname, 'cli/'),
```
will bundle our cli to cli folder in the root of the project

Comment (ajthinking) [reply webpack.config.js]: Good idea! This will do for now

**Comment by ajthinking** [comment]
> Keeping build artifacts in source control is really not a best practice
True, agree! Though I dont understand how gui access the files if there is no libs dir? The core package.json lets gui now it can find its files in lib?? :
```
  "files": [
    "lib/**/*"
  ],
```

> We also need to include cli folder in eslint and prettier ignore-files so as not to do extra work by checking a lot of code.
Fixed!

**Comment by Lenivaya** [comment]
> Though I dont understand how gui access the files if there is no libs dir? 

Well, because there is a lib dir. It is created by life-cycle scripts defined in package.json, specifically prepare script which runs on npm install or yarn install. [Here are some more information](https://docs.npmjs.com/cli/v7/using-npm/scripts#life-cycle-scripts)

**Comment by ajthinking** [comment]
> Well, because there is a lib dir. It is created by life-cycle scripts defined in package.json, specifically prepare script which runs on npm install or yarn install. Here are some more information

Aaaah got it

**Pull request merged**